TPP.Model/BadgeBuyOffer.cs

```diff
@@ -0,0 +1,90 @@
+using NodaTime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TPP.Common;
+
+namespace TPP.Model
+{
+    public class BadgeBuyOffer
+    {
+        /// <summary>
+        /// Unique Id.
+        /// </summary>
+        public string Id { get; init; }
+
+        /// <summary>
+        /// The ID of the user that created the buy offer.
+        /// </summary>
+        public string UserId { get; init; }
+
+        /// <summary>
+        /// The species of pokemon to buy.
+        /// </summary>
+        public PkmnSpecies Species { get; init; }
+
+        /// <summary>
+        /// The form of pokemon to buy.
+        /// </summary>
+        public string? Form { get; init; }
+
+        /// <summary>
+        /// The source of the badge to buy.
+        /// </summary>
+        public Badge.BadgeSource? Source { get; init; }
+
+        /// <summary>
+        /// Is the offer seeking shiny badges.
+        /// </summary>
+        public bool? Shiny { get; init; }
+
+        /// <summary>
+        /// How much to pay for each badge.
+        /// </summary>
+        public int Price { get; init; }
+
+        /// <summary>
+        /// The number of badges to buy.
+        /// </summary>
+        public int Amount { get; private set; }
+
+        /// <summary>
+        /// When the buy offer was created.
+        /// </summary>
+        public Instant CreatedAt { get; init; }
+
+        /// <summary>
+        /// When this offer was last updated.
+        /// </summary>
+        public Instant WaitingSince { get; private set; }
+
+        //duration and expires_at depricated from old core
+
+        public BadgeBuyOffer(string id, string userId, PkmnSpecies species, string? form, Badge.BadgeSource? source, bool? shiny, int price, int amount, Instant createdAt)
+        {
+            Id = id;
+            UserId = userId;
+            Species = species;
+            Form = form;
+            Source = source;
+            Shiny = shiny;
+            Price = price;
+            Amount = amount;
+            CreatedAt = createdAt;
+            WaitingSince = createdAt;
```

Comment:
`WaitingSince = createdAt` even if just a little bit, but this looks like it puts a bit of logic into a data class. I think it's better for this to be passed in as well, and the caller can then decide to use the same date twice.